oddt/virtualscreening.py

```diff
@@ -204,13 +218,12 @@ def similarity(self, method, query, cutoff=0.9, protein=None):
             dist = usr_similarity
         else:
             raise ValueError('Similarity filter "%s" is not supported.' % method)
-        query_fps = [(gen(q) if protein is None else gen(q, protein))
-                     for q in query]
-        self._pipe = filter(lambda q: any(dist(gen(q) if protein is None
-                                               else gen(q, protein),
-                                               q_fp) >= float(cutoff)
-                                          for q_fp in query_fps),
-                            self._pipe)
+        query_fps = [gen(q) for q in query]
```

Comment:
Add a comments here and in L224, smth like "generate FPs for query molecules" + "use the same FP generator for molecules in pipe".